packages/core/src/router.ts

```diff
@@ -495,24 +496,18 @@ export class Router {
 
   protected pushState(page: Page): void {
     this.page = page
-    if (isChromeIOS) {
+    if (isChromeIOS || isEdgeIOS) {
       // Defer history.pushState to the next event loop tick to prevent timing conflicts.
       // Ensure any previous history.replaceState completes before pushState is executed.
-      setTimeout(() => window.history.pushState(cloneSerializable(page), '', page.url))
+      setTimeout(() => window.history.pushState(cloneSerializable(page), '', page.url), 1)
     } else {
       window.history.pushState(cloneSerializable(page), '', page.url)
     }
   }
 
   protected replaceState(page: Page): void {
     this.page = page
-    if (isChromeIOS) {
-      // Defer history.replaceState to the next event loop tick to prevent timing conflicts.
-      // Ensure any previous history.pushState completes before replaceState is executed.
-      setTimeout(() => window.history.replaceState(cloneSerializable(page), '', page.url))
-    } else {
-      window.history.replaceState(cloneSerializable(page), '', page.url)
-    }
```

Comment:
Curious, why did you delete this block of code? Wouldn't it just need to be updated to `isChromeIOS || isEdgeIOS` as well?

Reply:
During my testing, I observed that when navigating from one page to another, the replaceState function is called first, followed by pushState, and then replaceState is triggered again. For example, when transitioning from the home page to the dashboard, the browser would first process the home page with replaceState, then execute pushState for the dashboard page, and afterward call replaceState again. With this sequence, the back navigation worked, but I had to click the back button twice to return to the previous page. After adjusting the replaceState function, the issue of adding the same page twice was resolved, and now a single back button click is enough to return to the previous page.